Add spring cleaning results blog post (#3203)
nf-core/website: sites/main-site/src/content/blog/2025/springcleaning.mdx

---
title: "Spring cleaning"
subtitle: "Keeping you informed of the latest maintainers actions"
pubDate: 2025-03-28T19:00:00+01:00
headerImage: "/assets/images/blog/springcleaning2025/kungstradgarden_cherry_tree.jpg"
headerImageAlt: "Spring in Stockholm with the cherry trees from Kungsträdgården - Picture by @maxulysse 2021-04-27"
headerImageDim: [250, 250]

> embedHeaderImage: true

Reviewer: ```suggestion
embedHeaderImage: false
```
looks too tiny otherwise imo

In this blog post we want to briefly summarise the outcome of the spring cleaning, and describe some of the proposals put forward by the maintainers team participants to improve our processes and procedures.

## Overview

Prior the spring cleaning, we split up into different groups that were focused on particular tasks:

- nf-core/pipelines
- nf-core/modules
- nf-core/documentation
- nf-core/configs
- Hackathon preparation
- Core Team

The teams were extremely productive throughout the week, as can be seen just with merged PRs alone from this [pulpo.io](https://pullpo.io/) graph:

![Bar chart of number of PRs merged by the maintainers team between March 10th-16th, with a total of 77 merged PRs throughout the week](src/assets/images/blog/springcleaning2025/maintainers-team-merged-stats.png)

## Progress

### Pipelines

The pipelines cleaning team were tasked with checking the state of each pipeline's repository.
This involved assessing the number of stale issues, branches, PRs etc, as well as checking minor things such as tags and descriptions were all present and correct.

> The pipelines team were also evaluating the development status of the pipeline, and if a pipeline seemed to be very 'quiet', began to contact the lead developers whether the pipeline is still under active development or maintainence.

Reviewer 2: ```suggestion
The pipelines team were also evaluating the development status of the pipeline, and if a pipeline seemed to be very 'quiet', began to contact the lead developers whether the pipeline is still under active development or maintenance.
```

By the end of the spring cleaning, almost half of the nf-core pipelines were assessed, most of them were in good shape.
Fun fact: of the 50 assessed, 5 were still written in Nextflow DSL1!

> Around 10 of them appear to be no longer maintained, and this list was passed to the the core team for further evaluation about their status and whether they should be possible archived.

Reviewer 2: ```suggestion
Around 10 of them appear to be no longer maintained, and this list was passed to the the core team for further evaluation about their status and whether they should be possibly archived.
```

A yearly check seemed to be enough even if not all pipelines were checked this time round.

> For next year's spring cleaning, we will ensure the team has more maintainers to work on this task next year.

Reviewer 2: ```suggestion
For next year's spring cleaning, Jonas and Florian requested we ensure the team has more maintainers to work on this task next year.
```

For example, we discussed potentially making better use of GitHub tags and [types](https://docs.github.com/en/issues/tracking-your-work-with-issues/configuring-issues/managing-issue-types-in-an-organization), to help both prepare spring-cleaning task boards, but also potentially to 'auto close' stale PRs and issues after a certain amount of time.
We also discussed maybe requiring all new modules proposals/issues to be made on pipeline repositories and linked to the modules repository, to ensure the module has a direct application, and thus the repository issues does not get filled up with just 'ideas'.
The modules maintainers maestros also pointed out there were inconsistencies in some places where modules were not following nf-core guidelines, so potentially we will ask the infrastructure team to add more linting rules.
They also suggested a good hackathon task to bring older nf-core subworkflows up to speed to the latest [specifications](https://nf-co.re/docs/guidelines/components/subworkflows).

### Configs

In comparison, the nf-core/configs team of Maxime, Joon, and James had an easier time due to the smaller nature of the repository.

> During the week, 4 issues were closed and,  9 the authors were bumped, 17 stale branches deleted, and 7 PRs merged or closed with 10 outstanding.

Reviewer 2: I think that Joon/Maxime asked the original authors of teh issues iif it s still valid - I am tweaking the text to reflex it :)

> During the week, 4 issues were closed and, in 9 the authors were bumped, 17 stale branches deleted, and 7 PRs merged or closed with 10 outstanding.

Reviewer 2: ```suggestion
During the week, 4 issues were closed, in 9 issues the original authors were nudged for further information, 17 stale branches deleted, and 7 PRs merged or closed with 10 outstanding.
```

> So while regular clean up is done more or less regularly by James and Maxime, and this repo is smaller, so this was less of an effort - but thanks to Joon for going beyond his normal responsibilities for this one!

Reviewer 2: ```suggestion
So while regular clean up is done more or less regularly by James and Maxime, and this repo is smaller, it still was good to give it a good look through. Thanks to Joon for going beyond his normal responsibilities for this one!
```

> We did discuss setting up 'stale-bot' automation to remove abandoned PRs and issues - as there was a tendency of this to happen, but given the community volunteers don't have access to all HPCs, it is not something we can finalise.

Reviewer 2: ```suggestion
Despite the low-traffic nature of the repository, we did discuss the setting up 'stale-bot' automation to remove abandoned PRs and issues here as well - as there was a tendency of abandoned PRs/issues, but given the community volunteers don't have access to all HPCs, it is not something we can finalise.
```

### Hackathon preparation

> Nicolas lead a team to try and add more tasks and issues to the [March 2025 hackathon project board](https://github.com/orgs/nf-core/projects/99) - currently somewhere between 300-400 issues!

Reviewer 2: ```suggestion
Nicolas led a team to try and add more tasks and issues to the [March 2025 hackathon project board](https://github.com/orgs/nf-core/projects/99) - currently somewhere between 300-400 issues!
```

This is still an ongoing project, and we will continue asking people to add tasks to project board!
So keep adding what you plan to work on - or if you are looking for help on something, add an issue so community volunteers can help you out!

### Core team

The core team did their routine checks and overhauls of our ['internal' documentation](https://nf-co.re/docs/checklists/community_governance/core_team), and other places that only people with the keys can clean up.

> The GitHub and Slack were checked and updated where necessary, as well as the core-team and maintainer-checklists by nicolas.

Reviewer 2: ```suggestion
The GitHub and Slack were checked and updated where necessary, as well as the core-team and maintainer-checklists by Nicolas.
```

> Issues with the AWS megatests on the nf-core's Seqera Platform workspace were investigated, and all older and broken computing environments were been cleared by Rike.

Reviewer 2: ```suggestion
Issues with the AWS megatests on the nf-core Seqera Platform workspace were investigated, and all older and broken computing environments were been cleared by Rike.
```

All recent [proposals](https://nfcore.slack.com/archives/C06L02SNLVA) for [special interest groups](https://nf-co.re/special-interest-groups) were reviewed and website PRs opened by Jose - if you're interested in 'Core facilities', 'Cancer Genomics', or 'Immunology' then you might be interested in these newly accepted groups!
Maxime went through all slack channels and closed unused channels.
We also reviewed over 100 new pipeline proposals from the last 3 years! Of these, 37 proposals were moved to 'timed out', and 16 accepted proposals were flagged for archiving due to a lack of development.

> We discussed the current new pipelines proposal procedure, as we had many abandoned proposals and also the current GitHub board for it is not really copying with the history of 200 proposals.

Reviewer 2: ```suggestion
We discussed the current new pipelines proposal procedure, as we had many abandoned proposals and also the current GitHub board used for tracking them is not really coping with the history of >200 proposals.
```

subtitle: "🧹"
pubDate: 2025-03-20T12:00:00+01:00
headerImage: "/assets/images/blog/springcleaning2025/kungstradgarden_cherry_tree.jpg"
headerImageAlt: "Spring in Stockholm with the cherry trees from Kungsträdgården - Picture by @maxulysse 2021-04-27"
headerImageDim: [250, 250]
embedHeaderImage: false
authors:
    - "maxulysse"
    - "jfy133"
label:
    - "maintainers"
---

import Profile from "@components/GitHubProfilePictureExtended.svelte";
import { Image } from "astro:assets";

Last week, the maintainers team volunteered their time on behalf of the community to spring clean the whole common nf-core ecosystem in preparation for the upcoming hackathon!

**We would first like to thank everyone involved**, who helped to clean many of the nf-core github repositories by closing issues, updating, merging, or closing PRs.
It was a tremendous effort. Thanks to all of you, the state of nf-core is in a much better and cleaner state than before - and just in time for our largest hackathon ever!

In this blog post we want to briefly summarise the outcome of the spring cleaning, and describe some of the proposals put forward by the maintainers team to improve our processes and procedures.

> ## Overview

Reviewer: ```suggestion
## Organisation
```

🛠️ A yearly check seemed to be enough even if not all pipelines were checked this time round.
For next year's spring cleaning, Jonas and Florian requested we ensure the team has more maintainers to work on this task.

### Modules

📑 The modules team were tasked with tackling the 🦣 modules repository of more than 1,400 modules, and hundreds of open issues and PRs.

✅ An **astounding** number of issues and PRs were reviewed and closed.
Throughout the week this team reviewed 162 PRs, closing at least 62 of them, they also reviewed 213 issues, closing 72.
Special shout-out to the MVPs of Famke (52 PRs and 108 issues!), Luisa, and Simon who started _early_ and powered through the huge number of issues and PRs.

🛠️ During our wrap up session we discussed a number of options to better handle the modules repository using automation.
For example, we discussed potentially making better use of GitHub tags and [types](https://docs.github.com/en/issues/tracking-your-work-with-issues/configuring-issues/managing-issue-types-in-an-organization), to help both prepare spring-cleaning task boards, but also to potentially 'auto close' stale PRs and issues after a certain amount of time.

> We also discussed maybe requiring all new modules proposals/issues to be made on pipeline repositories and linked to the modules repository, to ensure the module has a direct application, and thus the repository issues does not get filled up with just 'ideas'.

Reviewer: Just a comment on the discussed point here: nf-core modules go now beyond nf-core pipelines, so this would hinder the spread of our standards and lead to more split up efforts.

Reviewer 2: We can discuss in the next maintainers meeting :)
